### PR TITLE
Added logic to write output from the SQL engine directly into the BigQuery sink.

### PR DESCRIPTION
**Pull request by fernst**

### CODE REVIEW

File: src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/BigQuerySQLEngine.java [hunk 1]
```diff
@@ -318,6 +338,115 @@ public SQLDatasetProducer getProducer(SQLPullRequest pullRequest, PullCapability
...
+    Dataset destDataset = bigQuery.getDataset(destinationDatasetId);
+    if (srcDataset != null && destDataset != null
+      && !Objects.equals(srcDataset.getLocation(), destDataset.getLocation())) {
+      LOG.warn("Direct table copy is only supported if both datasets are in the same location.");
+      return SQLWriteResult.unsupported(datasetName);
+    }
+
+    Table destTable = bigQuery.getTable(destinationTableId);
+
+    // TODO(fernst): Add logic to create the table if it doesn't exist.
```
Comment (tivv) [inline src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/BigQuerySQLEngine.java]: Please put JIRA number to any TODO

File: src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/BigQuerySQLEngine.java [hunk 2]
```diff
@@ -318,6 +338,115 @@ public SQLDatasetProducer getProducer(SQLPullRequest pullRequest, PullCapability
...
+    result = queryJob.getQueryResults();
+
+    // Check for errors
+    if (queryJob.getStatus().getError() != null) {
+      LOG.error("Error executing BigQuery Job: '{}' in Project '{}', Dataset '{}': {}",
+                jobId, project, dataset, queryJob.getStatus().getError().toString());
+      return SQLWriteResult.faiure(datasetName);
+    }
+
+    LOG.info("Copied data from {} to {}", sourceTableName, destinationTableName);
```
Comment (tivv) [inline src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/BigQuerySQLEngine.java]: nit: include number of rows

**Review by tivv — APPROVED** [review]
Two minor comments, otherwise LGTM

**Review by sanjanasandeep — APPROVED** [review]
LGTM

**Pull request merged**